### PR TITLE
Req6 superset of PR #7

### PR DESCRIPTION
**Pull request by jyutzler**

### CODE REVIEW

File: spec/1_related_tables.adoc [hunk 1]
```diff
@@ -119,30 +119,41 @@ For each row in `gpkgext_relations` there SHALL be a table or view of the name r
 [caption=""]
 .Requirement 6
 ====
-For user-defined media tables as referenced in `gpkgext_relations`, if the value of its foreign column (as specified in <<gpkgext_relations_table>>) is not null, the corresponding user-defined primary table (specified by `gpkgext_relations`) SHALL contain a row with an `id` corresponding to the value in that foreign column.
+For any user-defined media table, as referenced in `gpkgext_relations`, if the value of its `foreign_column` (as specified in <<gpkgext_relations_table>>) is not null, the corresponding user-defined primary table (specified by `gpkgext_relations`) SHALL contain at least one a row with a value in its `primary_column` (specified by `gpkgext_relations`) equal to the value in the user-defined media table `foreign_column`.
```
Comment (bradh) [inline spec/1_related_tables.adoc]: "at least one a row" should be "at least one row"

File: spec/1_related_tables.adoc [hunk 2]
```diff
@@ -119,30 +119,41 @@ For each row in `gpkgext_relations` there SHALL be a table or view of the name r
...
-.media
+.media table values
+[width="80%",options="header"]
 |=======================================================================
-|id|data|content_type|foreign_column
+|id|data|content_type|relation
 |17|<BLOB>|image/png|7
+|18|<BLOB>|image/png|8
+|19|<BLOB>|image/png|8
 |=======================================================================
```
Comment (bradh) [inline spec/1_related_tables.adoc]: What happens if I now want to add a new feature (`feature_id` of 4), which relates to media with `id` of 17 and 19? What would go in the `relation` columns?

File: spec/1_related_tables.adoc [hunk 3]
```diff
@@ -119,30 +119,41 @@ For each row in `gpkgext_relations` there SHALL be a table or view of the name r
...
 
-For each row of media with a non-null foreign_column, there must be a row in features with a matching id in the  primary_column.
+For each row of media with a non-null `foreign_column`, there must be at least one row in `features` with a matching id in the `primary_column`.
```
Comment (bradh) [inline spec/1_related_tables.adoc]: should "`foreign_column`" be "`relation` column" here?
should "`primary_column`" be "`relation` column" here?

**Comment by jyutzler** [comment]
@rcasscslt Please take a look at Brad's comments.

File: spec/1_related_tables.adoc (hunk 1 above)
Comment (rcasscslt) [reply spec/1_related_tables.adoc]: Yep

File: spec/1_related_tables.adoc (hunk 2 above)
Comment (rcasscslt) [reply spec/1_related_tables.adoc]: Busted, we will need a mapping table. I will come up with the changes for this for review.

File: spec/1_related_tables.adoc (hunk 3 above)
Comment (rcasscslt) [reply spec/1_related_tables.adoc]: I see what you're getting at. They were chosen from their names in the extension row. The primary column being in the features and the foreign column being in the media table.  I'll see what I can do about the wording there.

**Comment by rcasscslt** [comment]
Was not "watching" this repo. Am now.

**Comment by rcasscslt** [comment]
Jeff,

Can we have sub requirements e.g. 1.1?

**Comment by rcasscslt** [comment]
Seeing as we may be enhancing the definition of the relationships, does anyone else think it's a good idea to name the relationship type one-to-many etc.? Knowing the relationship type may help to guide strategies for reading and writing the data.

**Comment by bradh** [comment]
I don't think that naming is quite right. I could see value in constraining the relationship (e.g. to support a domain profile that says the relationship is 1:1, 1:n or n:1 for certain table / column relationships).

I could also see value in being able to assign some semantics to the relationship ("belongs to", "is an example of", or something like that), but really not sure if we should just provide a mechanism that can be extended to do that, or if we should have something in the documentation to explain how, or even some specific implementation support (like identifying the constrained vocabulary table, plus a column constraint).

**Comment by jyutzler** [comment]
@rcasscslt if you create a new set of changes, can you branch off of my req6 branch so that the full set of changes (including 9227b8b) remains intact? Then you can make a new PR and we'll kill this one.

**Comment by jyutzler** [comment]
@rcasscslt Generally we don't do subrequirements. Just make new ones.

**Comment by rcasscslt** [comment]
Thanks,

I was wondering how to keep the chain intact.

Rob

On November 3, 2017 11:22:21 AM NDT, Jeff Yutzler <notifications@github.com> wrote:
>@rcasscslt if you create a new set of changes, can you branch off of my
>req6 branch so that the full set of changes (including 9227b8b) remains
>intact? Then you can make a new PR and we'll kill this one.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/jyutzler/geopackage-related-tables/pull/8#issuecomment-341708948

-Rob

**Comment by rcasscslt** [comment]
K thx

On November 3, 2017 11:23:54 AM NDT, Jeff Yutzler <notifications@github.com> wrote:
>@rcasscslt Generally we don't do subrequirements. Just make new ones. 
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/jyutzler/geopackage-related-tables/pull/8#issuecomment-341709311

-Rob

**Pull request merged**